app/src/main/java/com/bashkevich/androidfundamentals/model/EntityMapper.kt

```diff
@@ -1,40 +1,59 @@
 package com.bashkevich.androidfundamentals.model
 
 import com.bashkevich.androidfundamentals.model.dto.ActorDto
-import com.bashkevich.androidfundamentals.model.dto.GenreDto
 import com.bashkevich.androidfundamentals.model.dto.MovieDto
-import com.bashkevich.androidfundamentals.model.entity.Actor
-import com.bashkevich.androidfundamentals.model.entity.Genre
-import com.bashkevich.androidfundamentals.model.entity.Movie
+import com.bashkevich.androidfundamentals.model.entity.ActorEntity
+import com.bashkevich.androidfundamentals.model.entity.MovieEntity
+import com.bashkevich.androidfundamentals.model.entity.MovieWithActors
+import com.bashkevich.androidfundamentals.model.viewobject.Actor
+import com.bashkevich.androidfundamentals.model.viewobject.Movie
 
 object EntityMapper {
-    fun convertToMovieFromDto(movieDto: MovieDto): Movie {
 
-        return Movie(
-            id = movieDto.id,
+    fun toMovieEntity(movieDto: MovieDto) =
+        MovieEntity(
+            movieId = movieDto.id,
             title = movieDto.title,
-            overview = movieDto.overview,
-            poster = makeImageUrl(movieDto.posterPicture),
-            backdrop = makeImageUrl(movieDto.backdropPicture),
-            ratings = movieDto.ratings,
-            numberOfRatings = movieDto.votesCount,
-            minimumAge = if (movieDto.adult) 16 else 13,
+            posterPath = makeImageUrl(movieDto.posterPicture),
+            backdropPath = makeImageUrl(movieDto.backdropPicture),
             runtime = movieDto.runtime,
-            genres = movieDto.genres?.map { convertGenreFromDto(it) }
+            genres = movieDto.genres?.joinToString { it.name }?.filter { !it.isWhitespace() },
```

Comment:
Bug - filtering should be before joining? Also, you can use String.isBlank()

Reply:
Ah no, you're trying to get a string like "a,b,c". You can just use "," delimiter in the `joinToString`